.github/actions/setup-matrix/action.yml

```diff
@@ -0,0 +1,23 @@
+name: 'Setup Matrix'
+description: 'Setup a test matrix'
+
+inputs:
+  shards:
+    description: 'Number of shards'
+    required: false
+    default: '1'
+outputs:
+  shards:
+    description: 'The test matrix shards'
+    value: ${{ steps.set-matrix.outputs.shards }}
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/checkout@v2
```

Comment:
Why does this need to checkout the repo?

Reply:
You need to perform a checkout in order to use powershell.